### PR TITLE
Add a wrapBlock function to Text

### LINKED ISSUE #6719: Table layout in 3.x CLI messed up for WIN OS
This is the output of `cake bake` for example (CakePHP v3.0.6):
```
Subcommands:

all                 Bake a complete MVC skeleton.
behavior            Bake a behavior class file.
cell                Bake a cell class file.
component           Bake a component class file.
controller          Bake a controller skeleton.
fixture             Generate fixtures for use with the test suite. You
                    can use `bake fixture all` to bake all fixtures.
form                Bake a form class file.
helper              Bake a helper class file.
model               Bake table and entity classes.
plugin              Create the directory structure, AppController class
                    and testing setup for a new plugin. Can create plugins in an
y of your
                    bootstrapped plugin paths.
shell               Bake a shell class file.
template            Bake views for a controller, using built-in or
                    custom templates.
test                Bake test case skeletons for classes.
migration_snapshot  Bake migration snapshot class.
migration           Bake migration class.

To see help on a subcommand use `cake bake.bake [subcommand] --help`
```

The overflow of all 3 and more line parts is missing the indentation, here `y of your`.

### PR DESCRIPTION
**Pull request by antograssiot**
This function allow to ensure that a wrapped text won't be larger than its specified width even if it's indented internally.

Useful for console table display and fix #6719

### CODE REVIEW

**Comment by jadb** [comment]
:clap:

File: tests/TestCase/Utility/TextTest.php [hunk 1]
```diff
@@ -449,6 +449,56 @@ public function testWrapIndent()
 TEXT;
         $this->assertTextEquals($expected, $result);
     }
+    /**
```
Comment (ADmad) [inline tests/TestCase/Utility/TextTest.php]: Missing new line above. Same for other new methods too.

**Comment by antograssiot** [comment]
@ADmad updated

**Comment by dereuromark** [comment]
Nice :+1

File: src/Utility/Text.php [hunk 2]
```diff
@@ -312,6 +312,59 @@ public static function wrap($text, $options = [])
...
+        if (!empty($options['indent'])) {
+            $indentationLength = strlen($options['indent']);
+            $chunks = explode("\n", $wrapped);
+            $count = count($chunks);
+            if ($count < 2) {
+                return $wrapped;
+            }
+            $toRewrap = '';
+            for ($i = $options['indentAt']; $i < $count; $i++) {
+                $toRewrap .= substr($chunks[$i], $indentationLength) . ' ';
```
Comment (markstory) [inline src/Utility/Text.php]: You might want to use mb_substr() here. The user text could be multibyte.

**Comment by antograssiot** [comment]
@markstory I think it was ok to use ``substr`` as the value cames from ``strlen`` but I updated anyway and add also a test case for indentation with multibytes caracter.

**Comment by markstory** [comment]
substr() will probably be safe, but there could be annoying bugs where you slice a Katakana character in half.

**Comment by markstory** [comment]
Nice work @antograssiot

**Pull request merged**